src/server/src/routes/facultyMember.ts

```diff
@@ -82,24 +84,34 @@ router.post(
  *          GET Request - Read - "GET /api/facultyMember/get-profile"
  ******************************************************************************/
 
-router.get('/get-profile', async (req: Request, res: Response) => {
-    const { id } = req.body;
-    try {
-        if (!id) {
-            return res.status(BAD_REQUEST).json({
-                error: errors.paramMissingError,
-            });
+router.get('/get-profile',
+    passport.authenticate('jwt', { session: false }),
+    async (req: Request, res: Response) => {
+        const { userId } = req.user as JWTUser;
```

Comment:
No need to check if there is an userId. As long as you put **passport.authenticate('jwt', { session: false })** in the middleware section of the request (like you did), the route will be protected and internally return an UNAUTHORIZED status if the user is not logged in.

Reply:
Done.